src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import * as styledEl from './styled'
+import { CurrencySelectButton } from '@cow/modules/swap/pure/CurrencySelectButton'
+import { formatSmartAmount } from 'utils/format'
+import { FiatValue } from 'components/CurrencyInputPanel/FiatValue'
+import { Trans } from '@lingui/macro'
+import { PriceImpact } from 'hooks/usePriceImpact'
+import { CurrencyInfo } from '@cow/common/pure/CurrencyInputPanel/typings'
+
+interface BuiltItProps {
+  className: string
+}
+
+export interface CurrencyPreviewProps extends Partial<BuiltItProps> {
+  id: string
+  currencyInfo: CurrencyInfo
+  priceImpactParams?: PriceImpact
+  topLabel?: string
+}
+
+export function CurrencyPreview(props: CurrencyPreviewProps) {
+  const { id, currencyInfo, className, priceImpactParams, topLabel } = props
+  const { priceImpact, loading: priceImpactLoading } = priceImpactParams || {}
+  const { currency, balance, fiatAmount, viewAmount } = currencyInfo
+
+  return (
+    <>
+      <styledEl.Wrapper id={id} className={className} withReceiveAmountInfo={false}>
+        {topLabel && <styledEl.CurrencyTopLabel>{topLabel}</styledEl.CurrencyTopLabel>}
+
+        <styledEl.CurrencyInputBox flexibleWidth={true}>
+          <div>
+            <CurrencySelectButton currency={currency || undefined} loading={false} readonlyMode={true} />
+          </div>
+          <div>
+            <styledEl.NumericalInput
+              className="token-amount-input"
+              readOnly={true}
+              value={viewAmount}
+              onUserInput={() => void 0}
```

Comment:
Hmm?

Reply:
We have `readOnly={true}` above, however, the `NumericalInput` component interface require `onUserInput` prop and i just added empty function